diff --git a/arch/arm64/boot/dts/mediatek/mt8192-asurada.dtsi b/arch/arm64/boot/dts/mediatek/mt8192-asurada.dtsi
index 32d7ae021..3fabd7c7f 100644
--- a/arch/arm64/boot/dts/mediatek/mt8192-asurada.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt8192-asurada.dtsi
@@ -273,47 +273,6 @@ sound: sound {
 		pinctrl-23 = <&aud_dat_miso_ch34_on_pins>;
 		pinctrl-24 = <&aud_gpio_tdm_off_pins>;
 		pinctrl-25 = <&aud_gpio_tdm_on_pins>;
-
-		audio-routing =
-			"Headphone Jack", "HPOL",
-			"Headphone Jack", "HPOR",
-			"IN1P", "Headset Mic",
-			"Speakers", "Speaker";
-
-		spk-playback-dai-link {
-			link-name = "I2S3";
-			dai-format = "i2s";
-			mediatek,clk-provider = "cpu";
-			codec {
-				sound-dai = <&rt1015p>;
-			};
-		};
-
-		hs-playback-dai-link {
-			link-name = "I2S8";
-			dai-format = "i2s";
-			mediatek,clk-provider = "cpu";
-			codec {
-				sound-dai = <&rt5682 0>;
-			};
-		};
-
-		hs-capture-dai-link {
-			link-name = "I2S9";
-			dai-format = "i2s";
-			mediatek,clk-provider = "cpu";
-			codec {
-				sound-dai = <&rt5682 0>;
-			};
-		};
-
-		displayport-dai-link {
-			link-name = "TDM";
-			dai-format = "dsp_a";
-			codec {
-				sound-dai = <&anx_bridge_dp>;
-			};
-		};
 	};
 };
 
