diff --git a/src/core/exec-invoke.c b/src/core/exec-invoke.c
index 829e094f519b..1d83440ff7cf 100644
--- a/src/core/exec-invoke.c
+++ b/src/core/exec-invoke.c
@@ -4,6 +4,7 @@
 #include <sys/ioctl.h>
 #include <sys/mount.h>
 #include <sys/prctl.h>
+#include <sys/file.h>
 
 #if HAVE_PAM
 #include <security/pam_appl.h>
diff --git a/src/firstboot/firstboot.c b/src/firstboot/firstboot.c
index 652a6d5c89fe..74321d769a86 100644
--- a/src/firstboot/firstboot.c
+++ b/src/firstboot/firstboot.c
@@ -4,6 +4,7 @@
 #include <getopt.h>
 #include <linux/loop.h>
 #include <unistd.h>
+#include <sys/file.h>
 
 #include "sd-id128.h"
 
diff --git a/src/network/networkd-sysctl.c b/src/network/networkd-sysctl.c
index 8fa0ede5c2ae..8298b79a7b6e 100644
--- a/src/network/networkd-sysctl.c
+++ b/src/network/networkd-sysctl.c
@@ -2,7 +2,7 @@
 
 #include <netinet/in.h>
 #include <linux/if.h>
-#include <linux/if_arp.h>
+//#include <linux/if_arp.h>
 
 #include "missing_network.h"
 #include "networkd-link.h"
diff --git a/src/portable/portable.c b/src/portable/portable.c
index 6054f0f17f8d..9a63acdc1b20 100644
--- a/src/portable/portable.c
+++ b/src/portable/portable.c
@@ -1,6 +1,7 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
 #include <linux/loop.h>
+#include <sys/file.h>
 
 #include "sd-messages.h"
 
diff --git a/src/shared/dev-setup.c b/src/shared/dev-setup.c
index 08fdd6057047..37e9369c03e6 100644
--- a/src/shared/dev-setup.c
+++ b/src/shared/dev-setup.c
@@ -3,6 +3,7 @@
 #include <errno.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <sys/file.h>
 
 #include "alloc-util.h"
 #include "dev-setup.h"
diff --git a/src/udev/udev-builtin-net_id.c b/src/udev/udev-builtin-net_id.c
index 248243600625..b7afd2ff0d36 100644
--- a/src/udev/udev-builtin-net_id.c
+++ b/src/udev/udev-builtin-net_id.c
@@ -39,6 +39,7 @@
 #include "strv.h"
 #include "strxcpyx.h"
 #include "udev-builtin.h"
+#include "missing_stdlib.h"
 
 #define ONBOARD_14BIT_INDEX_MAX ((1U << 14) - 1)
 #define ONBOARD_16BIT_INDEX_MAX ((1U << 16) - 1)
