Index: drivers/leds/leds-s2mu106.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/drivers/leds/leds-s2mu106.c b/drivers/leds/leds-s2mu106.c
--- a/drivers/leds/leds-s2mu106.c	(revision 5eeafe2bfcb93847b403fd92bd92f9d0a9f6341d)
+++ b/drivers/leds/leds-s2mu106.c	(date 1644158930000)
@@ -1058,6 +1058,7 @@
 
 int create_flash_sysfs(void)
 {
+#ifdef CONFIG_VIDEO_EXYNOS_FIMC_IS2
 	int err = -ENODEV;
 
 	if (IS_ERR_OR_NULL(camera_class)) {
@@ -1080,7 +1081,7 @@
 		pr_err("flash_sysfs: failed to create device file, %s\n",
 				dev_attr_rear_torch_flash.attr.name);
 	}
-
+#endif
 	return 0;
 }
 
@@ -1166,8 +1167,10 @@
 {
 	device_remove_file(flash_dev, &dev_attr_rear_flash);
 	device_remove_file(flash_dev, &dev_attr_rear_torch_flash);
+#ifdef CONFIG_VIDEO_EXYNOS_FIMC_IS2
 	device_destroy(camera_class, 0);
 	class_destroy(camera_class);
+#endif
 	mutex_destroy(&g_fled_data->lock);
 	return 0;
 }

