Subject: [PATCH] rem android ifdef
---
Index: drivers/usb/gadget/function/f_conn_gadget.ioctl.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/drivers/usb/gadget/function/f_conn_gadget.ioctl.h b/drivers/usb/gadget/function/f_conn_gadget.ioctl.h
--- a/drivers/usb/gadget/function/f_conn_gadget.ioctl.h	(revision 5eeafe2bfcb93847b403fd92bd92f9d0a9f6341d)
+++ b/drivers/usb/gadget/function/f_conn_gadget.ioctl.h	(date 1644158930000)
@@ -25,8 +25,6 @@
 	CONN_GADGET_IOCTL_BIND_STATUS_UNBIND = 2
 };
 
-#if defined(__ANDROID__)||defined(__TIZEN__)
-
 enum { 
 	CONN_GADGET_IOCTL_NR_0 = 0,
 	CONN_GADGET_IOCTL_NR_1,
@@ -43,5 +41,3 @@
 #define CONN_GADGET_IOCTL_BIND_GET_STATUS	_IOR(CONN_GADGET_IOCTL_MAGIC_SIG, CONN_GADGET_IOCTL_NR_2, int)
 #define CONN_GADGET_IOCTL_MAX_NR			CONN_GADGET_IOCTL_NR_MAX
 #endif
-
-#endif
