From 53d14f0e8b9d68050dceccbecc8219d356437c07 Mon Sep 17 00:00:00 2001
From: Eray Tutmaz <tarkanttmz261@gmail.com>
Date: Thu, 19 Oct 2023 08:31:12 +0000
Subject: [PATCH 3/6] Update file proc.S

---
 arch/arm64/mm/proc.S | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm64/mm/proc.S b/arch/arm64/mm/proc.S
index 4c2d56b9..7d2915bc 100644
--- a/arch/arm64/mm/proc.S
+++ b/arch/arm64/mm/proc.S
@@ -140,8 +140,6 @@ ENTRY(cpu_do_switch_mm)
 	b	post_ttbr_update_workaround	// Back to C code...
 ENDPROC(cpu_do_switch_mm)
 
-	.section ".text.init", #alloc, #execinstr
-
 /*
  *	__cpu_setup
  *
-- 
2.34.1

