From 5c21a20cc9d4c6f19b5d4baa1ac4ec40ce26ae0f Mon Sep 17 00:00:00 2001
From: Hendra Manudinata <manoedinata@gmail.com>
Date: Tue, 7 May 2024 23:42:50 +0700
Subject: [PATCH 5/6] staging: samsung: sec_batt: Force disable low power
 charge

Signed-off-by: Hendra Manudinata <manoedinata@gmail.com>
---
 drivers/staging/samsung/sec_batt.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/staging/samsung/sec_batt.c b/drivers/staging/samsung/sec_batt.c
index de9f63ef..5145dcc6 100644
--- a/drivers/staging/samsung/sec_batt.c
+++ b/drivers/staging/samsung/sec_batt.c
@@ -20,8 +20,11 @@ EXPORT_SYMBOL(lpcharge);
 
 static int sec_bat_is_lpm_check(char *str)
 {
+/* HACK: Force disable low power charge */
+#if 0
 	if (strncmp(str, "charger", 7) == 0)
 		lpcharge = 1;
+#endif
 
 	pr_info("%s: Low power charging mode: %d\n", __func__, lpcharge);
 
-- 
2.34.1

