apk-tools 2.12.10 fails with this in a .pc depends file
diff --git a/meson.build b/meson.build
index e29c5001bcad..3f21f10cf49a 100644
--- a/meson.build
+++ b/meson.build
@@ -25,9 +25,9 @@ gdk_pixbuf_req = '>= 2.0'
 pango_req = '>= 1.46.0'
 cairo_req = '>= 1.10.0'
 pangocairo_req = '>= 1.20'
 pixman_req = '>= 0.42'
-gsettings_desktop_schemas_req = '>= 40.alpha'
+gsettings_desktop_schemas_req = '>= 40_alpha'
 x11_req = '>= 1.7.0'
 xcomposite_req = '>= 0.4'
 xkbcommon_req = '>= 0.4.3'
 xfixes_req = '>= 6'
